eclipse-collections/src/main/java/org/eclipse/collections/impl/map/strategy/mutable/UnifiedMapWithHashingStrategy.java

```diff
@@ -1751,12 +1750,38 @@ public boolean containsAll(Collection<?> collection)
             return true;
         }
 
+        @Override
+        public void each(Procedure<? super K> procedure)
+        {
+            UnifiedMapWithHashingStrategy.this.forEachKey(procedure);
+        }
+
         @Override
         public boolean isEmpty()
         {
             return UnifiedMapWithHashingStrategy.this.isEmpty();
         }
 
+        @Override
+        public K getFirst()
+        {
+            if (UnifiedMapWithHashingStrategy.this.size() > 0)
+            {
+                return (K) this.toArray()[0];
+            }
+            return null;
+        }
+
+        @Override
+        public K getLast()
+        {
+            if (UnifiedMapWithHashingStrategy.this.size() > 0)
+            {
+                return (K) this.toArray()[UnifiedMapWithHashingStrategy.this.size() - 1];
```

Comment:
@johnjstuart I don't think that convering to array just to get the last element is a good idea, it's too costly IMHO

Reply:
Use `IterableIterate.getLast(this)`.

Reply:
Done 👍